src/python-common/ceph/deployment/service_spec.py

```diff
@@ -600,9 +600,15 @@ def __init__(self, service_id, pool=None,
         if not self.api_secure and self.ssl_cert and self.ssl_key:
             self.api_secure = True
 
-    def validate_add(self):
-        servicespec_validate_add(self)
+    def validate(self):
+        super(IscsiServiceSpec, self).validate()
 
         if not self.pool:
             raise ServiceSpecValidationError(
                 'Cannot add ISCSI: No Pool specified')
+        if not self.api_user:
+            raise ServiceSpecValidationError(
+                'Cannot add ISCSI: No Api user specified')
+        if not self.api_password:
+            raise ServiceSpecValidationError(
+                'Cannot add ISCSI: No Api password specified')
```

Comment:
this is going to get interesting if we're making an upgrade and users already have an iscsi service applied without user / password. 

Is there a super easy way to resolve this?